src/main/java/gov/cms/madie/models/cqm/datacriteria/AllergyIntolerance.java

```diff
@@ -28,4 +39,21 @@ public class AllergyIntolerance extends DataElement {
   private String qdmStatus = "intolerance";
   private String qdmVersion = "5.6";
   private String _type = "QDM::AllergyIntolerance";
+
+  public void shiftDates(int shifted) {
+
+    if (this.authorDatetime != null) {
+      this.authorDatetime = this.authorDatetime.plusYears(shifted);
+    }
+    if (this.prevalencePeriod != null) {
+      Interval changeInterval = this.prevalencePeriod;
+      if (changeInterval.getLow() != null) {
+        changeInterval.setLow(changeInterval.getLow().plusYears(shifted));
+      }
+      if (changeInterval.getHigh() != null) {
+        changeInterval.setHigh(changeInterval.getHigh().plusYears(shifted));
+      }
+      this.prevalencePeriod = changeInterval;
+    }
+  }
```

Comment:
a lot of code duplication across multiple files. was wondering if we could remove some duplication by introducing some utility methods maybe in Super class DataElement.
e.g.
`// DataElement.java`

```
public Interval shiftIntervalByYear(Interval interval, int year) {
    if (interval == null) {
      return null;
    }
    if (interval.getLow() != null) {
      interval.setLow(interval.getLow().plusYears(year));
    }
    if (interval.getHigh() != null) {
      interval.setHigh(interval.getHigh().plusYears(year));
    }
    return interval;
  }

  public ZonedDateTime shiftDateByYear(ZonedDateTime dateTime, int year) {
    if (dateTime == null) {
      return null;
    }
    return dateTime.plusYears(year);
  }
```
and then calling these methods from individual files.
e.g. in this file

```
public void shiftDates(int shifted) {
    this.authorDatetime = shiftDateByYear(this.authorDatetime, shifted);
    this.prevalencePeriod = shiftIntervalByYear(this.prevalencePeriod, shifted);
  } 
```
there will still be some duplicates but few lines less. other option is `getDeclaredFields`  of reflection API but I think that will cause some performance because we'll have to iterate over all of the properties to see if the property is an instance of date or interval

Reply:
@adongare Thanks! Will do.

Reply:
@adongare This is done, thanks!